api/src/clojure/org/akvo/flow_api/endpoint/sync.clj

```diff
@@ -41,20 +41,24 @@
                 db-spec (-> deps :unilog-db :spec (assoc :db-name db-name))
                 offset (Long/parseLong cursor)]
             (if (unilog/valid-offset? offset db-spec)
-              (let [changes (->
-                             (unilog/process-unilog-events offset db-spec instance-id (:remote-api deps))
-                             (select-keys [:form-instance-changed
-                                           :form-instance-deleted
-                                           :form-changed
-                                           :form-deleted
-                                           :data-point-changed
-                                           :data-point-deleted
-                                           :unilog-id])
-                             (update :form-deleted #(map str %))
-                             (update :form-instance-deleted #(map str %))
-                             (update :data-point-deleted #(map str %)))]
-                (response {:changes (dissoc changes :unilog-id)
-                           :next-sync-url (next-sync-url (utils/get-api-root req) alias (:unilog-id changes))}))
+              (if (= offset (unilog/get-cursor db-spec)) ;; end of the log
+                (-> (status {} 204)
+                    (header "Cache-Control" "max-age=3600"))
```

Comment:
Isn't the Nginx adding this header? (Haven't tested it)

In any case, there is some duplication here and in the nginx code regarding the caching time. Either we make the `proxy_cache_valid` follow this header or we remove this header.

Reply:
According to the documentation, nginx will honor the `Cache-Control` header from the upstream server. I'll double check what is the behavior when both properties differ.

Reply:
> Isn't the Nginx adding this header? (Haven't tested it)

nginx is not adding the header, but it does cache the HTTP 204 response. I can see the instrumentation header:
````
< HTTP/1.1 204 No Content
< Server: openresty
< Date: Tue, 28 Jan 2020 09:24:49 GMT
< Connection: keep-alive
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Credentials: true
< Access-Control-Allow-Methods: GET, OPTIONS
< Access-Control-Allow-Headers: Accept,Accept-Encoding,Accept-Language,Authorization,Connection,DNT,Host,Origin,Referer,User-Agent
< X-Cache-Status: MISS
< 

[...]

< HTTP/1.1 204 No Content
< Server: openresty
< Date: Tue, 28 Jan 2020 09:25:06 GMT
< Connection: keep-alive
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Credentials: true
< Access-Control-Allow-Methods: GET, OPTIONS
< Access-Control-Allow-Headers: Accept,Accept-Encoding,Accept-Language,Authorization,Connection,DNT,Host,Origin,Referer,User-Agent
< X-Cache-Status: HIT

````
In this case the server is not sending the `Cache-Control` but we do have the configuration ` proxy_cache_valid 204 <time>`